rust/rope/src/tree.rs

```diff
@@ -566,33 +579,74 @@ impl<'a, N: NodeInfo> Cursor<'a, N> {
         }
     }
 
+    // Moves the cursor to the next discontinuity, or to the end
+    // of the rope. In the former case, returns the offset of said
+    // discontinuity.
     pub fn next<M: Metric<N>>(&mut self) -> Option<(usize)> {
```

Comment:
hmm... I think it's still worse than that, since in the worst case we have to restart from the top when the cache is too small.